src/framework/mocha/Utilities/MOJSBlock.m

```diff
@@ -113,14 +113,18 @@ static JSValueRef jsInvoke(MOJavaScriptObject* function, NSMethodSignature* sign
         }
     }
     
-    JSContextRef ctx = [function JSContext];
-    Mocha *runtime = [Mocha runtimeWithContext:ctx];
-    
-    JSValueRef value = [runtime callJSFunction:[function JSObject] withArgumentsInArray:arguments];
+    __block JSValueRef value = NULL;
     
-    if (value == NULL) {
-        [NSException raise:@"CallbackError" format:@"Error while calling the callback"];
-    }
+    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
Also, please please can something like this have a comment noting why it's being done.